includes/class-wc-stripe-api.php

```diff
@@ -123,14 +123,31 @@ public static function get_headers() {
 		$user_agent = self::get_user_agent();
 		$app_info   = $user_agent['application'];
 
-		$headers = apply_filters(
+		$headers = apply_filters_deprecated(
```

Comment:
To simplify the eventual removal of the deprecated filter, it may be worth setting up `$headers` outside of the filter. (This is not blocking.)
```suggestion
		$headers = [
			'Authorization' => 'Basic ' . base64_encode( self::get_secret_key() . ':' ),
			'Stripe-Version' => self::STRIPE_API_VERSION,
		];

		$headers = apply_filters_deprecated(
```